wire.go

```diff
@@ -0,0 +1,68 @@
+package opentracing
+
+// WireEncoding identifies a WireInjector/WireExtractor pair.
+type WireEncoding int
+
+const (
```

Comment:
Note that OpenTracing-portable encodings will still need to be declared here if we do this as an enum.

Another approach – probably a *better* approach – would be to follow the `net.Context` approach which allows for a kind of "global type-namespace" of encodings without forcing anyone to deal with `package reflect`. Example: https://github.com/golang/blog/blob/master/content/context/userip/userip.go#L34